controllers/database.go

```diff
@@ -37,47 +38,49 @@ var mariadbTemplates = []string{
 }
 
 // extract to var for mocking in testing
-var ConnectAndQueryDatabase = func(host, port, username, password, dbname string, dbConnectionTimeout time.Duration) bool {
+var ConnectAndQueryDatabase = func(host, port, username, password, dbname string, dbConnectionTimeout time.Duration) (bool, string) {
```

Comment:
if the intended purpose of `message` is for error handling, why don't we just return `error`s rather than `string`s and let the caller of this function get the message from `errror.Error()`?  This is much closer to what is conventional in golang